src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,317 @@
+then iterate over them. Since everything is asynchronous, use the
+[`for ... of`][for-of] iterator, since it handles async/await code nicely.
+
+Each `ClipboardItem` can hold its contents in different types, so you'll
+need to iterate over the list of types, again using a `for ... of` loop.
+For each type, call the `getType()` method with the current type as an argument
+to obtain the corresponding image `Blob`. As before, this code is not tied
+to images, and will work with other future file types.
+
+```js
+async function getClipboardContents() {
+  try {
+    const clipboardItems = await navigator.clipboard.read();
+    for (const clipboardItem of clipboardItems) {
+      try {
+        for (const type of clipboardItem.types) {
+          const blob = await clipboardItem.getType(type);
+          console.log(URL.createObjectURL(blob));
+        }
+      } catch (e) {
+        console.error(e, e.message);
+      }
+    }
+  } catch (e) {
+    console.error(e, e.message);
+  }
+}
+```
+
+### Custom paste handler {: #custom-paste-handler }
+
+To dynamically handle paste events, listen for the `paste` event, call
+[`preventDefault()`][prevent-default] to prevent the default behavior
+in favor of your own logic, then use the code above to read the contents from
+the clipboard, and handle it in whatever way your app needs.
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  getClipboardContents();
+});
+```
+
+### Custom copy handler {: #custom-copy-handler }
+
+The [`copy` event][copy-event] includes a [`clipboardData`][clipboard-data]
+property with the items already in the right format, eliminating the need to
+manually create a blob. As before, don't forget to call `preventDefault()`.
+
+```js
+document.addEventListener('copy', async (e) => {
+  e.preventDefault();
+  try {
+    for (const item of e.clipboardData.items) {
+      await navigator.clipboard.write([
+        new ClipboardItem(Object.defineProperty({}, item.type, {
+          value: item,
+          enumerable: true
+        }))
+      ]);
+    }
+    console.log('Image copied.');
+  } catch(e) {
+    console.error(e, e.message);
+  }
+});
+```
+
+### Demo
+
+<iframe style="border: solid 1px;" width="100%" height="500"
+  src="https://jsfiddle.net/0794oysr/2/embedded/result,js,html,css"
+  allowfullscreen="allowfullscreen" frameborder="0">
+</iframe>
+
+### Security
+
+Opening up the Asynchronous Clipboard API for images comes with certain
+[risks](https://w3c.github.io/clipboard-apis/#security) that need to be
+carefully evaluated. One new challenge are so-called
+[image compression bombs](https://bomb.codes/bombs#images), that is, image
+files that appear to be innocent, but—once decompressed—turn out to be huge.
+Even more serious than large images are specifically crafted malicious images
+that are designed to exploit known vulnerabilities in the native operating
+system. This is why we can't just copy the image directly to the native
```

Comment:
```suggestion
system. This is why Chrome can't just copy the image directly to the native
```

Avoiding "we" per Rob's rationale here: https://github.com/GoogleChrome/web.dev/pull/1614#discussion_r333143069